packages/openapi-typescript/src/lib/ts.ts

```diff
@@ -80,6 +81,16 @@ export function addJSDocComment(schemaObject: AnnotatedSchemaObject, node: ts.Pr
     output.push(`@${field} ${String(serialized).replace(LB_RE, "\n *     ")}`);
   }
 
+  // anyOf
+  if (!schemaObject.description && Array.isArray(schemaObject.anyOf)) {
+    const anyOfDescriptions = schemaObject.anyOf
+      .map((subSchema: any) => subSchema.description)
+      .filter((description: string | undefined) => typeof description === "string" && description.trim() !== "");
+    if (anyOfDescriptions.length > 0) {
+      output.push(`@description ${anyOfDescriptions.join(" | ")}`);
+    }
+  }
```

Comment:
It seems like this will still fail for generating jsdoc descriptions for the other compositions `allOf` and `oneOf`.

At least in the case of `allOf` you could re-use this logic and intersect (`&`) the entries of instead of union-ing (`|`) them.